pkg/storaged/btrfs/volume.jsx

```diff
@@ -0,0 +1,208 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hopeg that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+import client from "../client";
+
+import { CardHeader, CardBody } from "@patternfly/react-core/dist/esm/components/Card/index.js";
+import { DescriptionList } from "@patternfly/react-core/dist/esm/components/DescriptionList/index.js";
+import { VolumeIcon } from "../icons/gnome-icons.jsx";
+
+import { new_card, new_page, get_crossrefs, ChildrenTable, PageTable, StorageCard, StorageDescription } from "../pages.jsx";
+import { StorageUsageBar, StorageLink } from "../storage-controls.jsx";
+import { fmt_size_long, validate_fsys_label, decode_filename } from "../utils.js";
+import { btrfs_usage, btrfs_is_volume_mounted } from "./utils.jsx";
+import { dialog_open, TextInput } from "../dialog.jsx";
+import { make_btrfs_subvolume_page } from "./subvolume.jsx";
+import { mounting_dialog } from "../filesystem/mounting-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+function volume_mount(volume) {
+    const block = client.blocks[volume.path];
+    mounting_dialog(client, block, "mount", null);
+}
+
+/*
+ * Udisks is a disk/block library so it manages that, btrfs turns this a bit
+ * around and has one "volume" which can have multiple blocks by a unique uuid.
+ *
+ * Cockpit shows Btrfs as following:
+ *
+ * -> btrfs subvolume
+ *    -> btrfs volume
+ *        -> btrfs device
+ *            -> block device
+ */
+export function make_btrfs_volume_card(parent, uuid) {
+    const block_devices = client.uuids_btrfs_blocks[uuid];
+    const block_btrfs = client.blocks_fsys_btrfs[block_devices[0].path];
+    const volume = client.uuids_btrfs_volume[uuid];
+    const use = btrfs_usage(client, volume);
+
+    const name = block_btrfs.data.label || uuid;
+    const btrfs_volume_card = new_card({
+        title: _("btrfs volume"),
+        next: null,
+        page_location: ["btrfs-volume", uuid],
+        page_name: name,
+        page_icon: VolumeIcon,
+        page_size: use[1],
+        component: BtrfsVolumePage,
+        props: { block_devices, uuid, use },
+    });
+
+    const actions = [];
+    const is_mounted = btrfs_is_volume_mounted(client, block_devices);
+    if (!is_mounted) {
+        actions.push({
+            title: _("Mount btrfs volume"),
```

Comment:
Why do we need this? I found it confusing, since the root/default subvolume is always in the list of subvolumes, no?

Reply:
We can drop it, it's untested but in theory power users could want to their fstab to be:
```
/dev/sda  / auto
```
This would mount the default btrfs subvolume, and with a tool like `snapper` they could also rollback by an older install by setting a different subvolume.

This is a power user thing so we can imo leave it out.

Reply:
I'll just drop it for now.